flink-streaming-java/src/test/java/org/apache/flink/streaming/api/DataStreamTest.java

```diff
@@ -906,6 +919,256 @@ public void testChannelSelectors() {
 	}
 
 	/////////////////////////////////////////////////////////////
+	// KeyBy testing
+	/////////////////////////////////////////////////////////////
+
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
```

Comment:
Well it may be more concise, but a try catch block makes it way more obvious that this part of the code throws an exception we care about.

try-catch blocks are also more commonin the codebase, and I'm wondering whether we want to introduce an inconsistency there.